### PR TITLE
DDF-4509 Unable to view workspace details when local catalog is disabled

### PR DESCRIPTION
**Pull request by Corey-Collins**
#### What does this PR do?
When local catalog is disabled you are unable to view workspace details.

Viewing details sends a /cql request to every source with federation set to enterprise. None of those (unless local catalog is enabled) is able to return the workspace metacard and results in a page of no results found when it should be showing workspace properties.

This allows the metacard component to send to the `local` UI node in the case you are viewing a workspace's details.
#### Who is reviewing it? 
@bellcc 
@bdeining 
@willwill96 
@GabrielFabian 

#### Ask 2 committers to review/merge the PR and tag them here.

@andrewkfiedler
@millerw8

#### How should this be tested?

- Ingest data into DDF
- Run query for that data and save a workspace
- Enable local catalog
- View workspace details
- See workspace details under table or inspector
- Turn local catalog off
- View workspace details again
- See that you retrieve workspace details again rather than "no results found"

#### What are the relevant tickets?
Fixes: #4509 

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by Corey-Collins** [comment]
@bakejeyner close to your second option. When `A` disables its local catalog and you go to Workspaces > View Workspace Details, it sends a `/cql` request for the workspace to `B` `C` and `D`. These sources are unable to retrieve a workspace metacard and you get the response "no results found". 
Instead this adds `local` to the metacard federation in the event you click 'View Workspace Details' so the UI node `E` can reach solr and get the workspace metacard, instead of `B` `C` and `D`

File: ui/packages/catalog-ui-search/src/main/webapp/component/metacard/metacard-federation.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ * version 3 of the License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+import { Model } from 'backbone'
+
+module.exports = new (Model.extend({
```
Comment (adimka) [inline ui/packages/catalog-ui-search/src/main/webapp/component/metacard/metacard-federation.js]: Our singleton instances tend to live in `ui/packages/catalog-ui-search/src/main/webapp/component/singletons`
We should probably define the model extension here and instantiate it in a small file there.

File: ui/packages/catalog-ui-search/src/main/webapp/component/metacard/metacard-federation.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  initialize: function() {
```
Comment (adimka) [inline ui/packages/catalog-ui-search/src/main/webapp/component/metacard/metacard-federation.js]: We can use lambda notation `() => {...}` here.

**Comment by andrewkfiedler** [comment]
This kind of feels like a workaround.

Right now, disabling the local catalog causes it to be removed during the source polling here: https://github.com/codice/ddf/blob/master/ui/packages/catalog-ui-search/src/main/webapp/js/model/Sources.js#L99

That means it never makes it into the collection of sources.  However, given we might end up storing things locally anyways (such as workspaces, searches, lists, etc), maybe we should reconsider how we're doing it.  

What we're doing at the moment makes it easy since we don't have to go to each component driven by the sources and have them deal with the disabling of the local catalog themselves, but I wonder if we're going to run into more scenarios where we actually want to search the local catalog (even if the user isn't supposed to know about it).  In the end, our easy choice might make things more complex.

I think I would prefer we change how we handle disabling the local catalog, because if I were to look at this code in the future I wouldn't understand why an enterprise query wouldn't encompass searching locally as well.  I'm wondering what others think about it though, @djblue ?

**Pull request closed without merging**